src/Microsoft.Azure.ServiceBus/Amqp/AmqpExceptionHelper.cs

```diff
@@ -155,6 +155,11 @@ static Exception ToMessagingContractException(string condition, string message,
                 return new QuotaExceededException(message);
             }
 
+            if (string.Equals(condition, AmqpErrorCode.MessageSizeExceeded.Value))
```

Comment:
`MessageSender` will throw `MessageSizeExceededException` when a message larger than max size is attempted to be sent. What's the scenario here?

Reply:
This is still just a client. The service can also throw this exception. 
It should ideally not happen, but its possible.

Reply:
You're saying a broker could throw an exception on a message large than the max size? Under what curcomstances that would happen?

Reply:
1. No matter how many client validations exist, broker validation is a MUST (due to different clients that we have).
1. Before the message reaches broker, some more routing information is generally added in between. Ideally it should all be accounted for already by giving some buffer to the message size (between the first node which gets your message, and the actual destination broker machines). But lets say there are some issues that creep up, either caused by added data on gateway or due to different message size evaluations between client and broker, or due to wrong max message size sent on link, etc, broker can end up throwing this exception. 
1. As mentioned earlier, these things should NOT happen. But we have seen it happening, so its a good defensive code to have.

Reply:
I did not think of option #2. 👍